### PR TITLE
network: create per-thread NMClient instance (#1931389)

### PR DESCRIPTION
**Pull request by rvykydal**

### CODE REVIEW

**Comment by rvykydal** [comment]
WIP for comments from Thomas Haller.

File: pyanaconda/modules/network/network.py [hunk 1]
```diff
@@ -68,17 +69,12 @@ def __init__(self):
...
-        if SystemBus.check_connection():
-            nm_client = NM.Client.new(None)
-            if nm_client.get_nm_running():
-                self.nm_client = nm_client
-                self.nm_client.connect("notify::%s" % NM.CLIENT_STATE, self._nm_state_changed)
-                initial_state = self.nm_client.get_state()
-                self.set_connected(self._nm_state_connected(initial_state))
-            else:
-                log.debug("NetworkManager is not running.")
+        self.nm_client = get_new_nm_client()
```
Comment (poncovka) [inline pyanaconda/modules/network/network.py:73]: Do you think this set up could be run later? Something like `_connect_to_hostname_service`.

**Review by poncovka — COMMENTED** [review]
Otherwise, it looks good to me.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me!

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me.

**Comment by jkonecny12** [comment]
I would appreciate some description in the comment of the commit message. Why we need that.

**Comment by t-feng** [comment]
hello, everybody
I got same problem.
this issue has any news? :)

**Comment by t-feng** [comment]
Create new NMClient instance for worker threads can fix this problem?
Applying this patch, we still got  _g_main_context_push_thread_default: assertion 'acquired_context' failed_ error. Because context is acquired in main thread.
Whether we should create new context for each worker threads？

**Comment by rvykydal** [comment]
I tried to create new context but it lead to hangs, see https://bugzilla.redhat.com/show_bug.cgi?id=1931389#c10

**Comment by t-feng** [comment]
@rvykydal hello
this issue has beed fixed by some certain PR?  Or fixed by high version? (anaconda-33.19 still has this problem with a very low probability)

**Pull request closed without merging**